mesheryctl/internal/cli/root/credentials/create.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Meshery Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package credentials
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var createCredentialCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a new credential",
+	Long:  `Create a new credential by providing the name, user ID, type, and secret of the credential`,
+	Example: `
+// Create a new credential
+mesheryctl exp credential create --user-id [user-id]
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return system.ErrGetCurrentContext(err)
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+
+	Args: func(cmd *cobra.Command, args []string) error {
+		user_id, _ = cmd.Flags().GetString("user-id")
+		name, _ = cmd.Flags().GetString("name")
+		credential_type, _ = cmd.Flags().GetString("type")
+		secret, _ = cmd.Flags().GetString("secret")
+
+		if user_id == "" || name == "" || credential_type == "" || secret == "" {
+			if err := cmd.Usage(); err != nil {
+				return err
+			}
+			return utils.ErrInvalidArgument(errors.New("user-id, name, type, and secret are required"))
+		}
+		return nil
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+
+		baseURL := mctlCfg.GetBaseMesheryURL()
+		url := fmt.Sprintf("%s/api/integrations/credentials", baseURL)
+
+		// Generate a unique identifier for the credential
+		id, err := uuid.NewV4()
+		if err != nil {
+			return err
```

Comment:
Check this function somewhere we are logging error and returning nil other places we are returning error. I wonder why? :thinking: